compose/config/config.py

```diff
@@ -165,13 +164,11 @@ def find_candidates_in_parent_dirs(filenames, path):
     return (candidates, path)
 
 
-@validate_top_level_object
-@validate_service_names
 def pre_process_config(config):
-    """
-    Pre validation checks and processing of the config file to interpolate env
+    """Pre validation checks and processing of the config file to interpolate env
     vars returning a config dict ready to be tested against the schema.
     """
+    validate_top_level_object(config)
```

Comment:
Yeh, I think given how the code has changed that it doesn't fit into the model of how things work now and the different places it needs to be called. So I'm +1 with it being dropped.